v2/cmd/musa/musa.go

```diff
@@ -0,0 +1,223 @@
+				Value:       cfg.Host,
+				Destination: &cfg.Host,
+				EnvVars:     []string{"MUSA_HOST"},
+			},
+			&cli.IntFlag{
+				Name:        "port",
+				Usage:       "the port on which musa should listen on",
+				Value:       cfg.Port,
+				Destination: &cfg.Port,
+				EnvVars:     []string{"MUSA_PORT"},
+				DefaultText: "random",
+			},
+			&cli.StringFlag{
+				Name:        "protocol",
+				Usage:       "the libp2p protocol for the DHT",
+				Value:       cfg.ProtocolID,
+				Destination: &cfg.ProtocolID,
+				EnvVars:     []string{"MUSA_PROTOCOL"},
+			},
+			&cli.StringFlag{
+				Name:        "metrics-host",
+				Usage:       "the network musa metrics should bind on",
+				Value:       cfg.MetricsHost,
+				Destination: &cfg.MetricsHost,
+				EnvVars:     []string{"MUSA_METRICS_HOST"},
+			},
+			&cli.IntFlag{
+				Name:        "metrics-port",
+				Usage:       "the port on which musa metrics should listen on",
+				Value:       cfg.MetricsPort,
+				Destination: &cfg.MetricsPort,
+				EnvVars:     []string{"MUSA_METRICS_PORT"},
+				DefaultText: "random",
+			},
+			&cli.StringFlag{
+				Name:        "trace-host",
+				Usage:       "the network musa trace should be pushed to",
+				Value:       cfg.TraceHost,
+				Destination: &cfg.TraceHost,
+				EnvVars:     []string{"MUSA_TRACE_HOST"},
+			},
+			&cli.IntFlag{
+				Name:        "trace-port",
+				Usage:       "the port to which musa should push traces",
+				Value:       cfg.TracePort,
+				Destination: &cfg.TracePort,
+				EnvVars:     []string{"MUSA_TRACE_PORT"},
+				DefaultText: "random",
+			},
+		},
+	}
+
+	sigs := make(chan os.Signal, 1)
+	ctx, cancel := context.WithCancel(context.Background())
+
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM, os.Interrupt, os.Kill)
+	go func() {
+		sig := <-sigs
+		slog.Info("Received signal - Stopping...", "signal", sig.String())
+		signal.Stop(sigs)
+		cancel()
+	}()
+
+	if err := app.RunContext(ctx, os.Args); err != nil {
+		slog.Error("application error", "err", err)
+		os.Exit(1)
+	}
+}
+
+func daemonAction(cCtx *cli.Context) error {
+	slog.Info("Starting musa daemon process with configuration:")
+	fmt.Println(cfg.String())
+
+	exporter, err := prometheus.New()
+	if err != nil {
+		return fmt.Errorf("new prometheus exporter: :%w", err)
+	}
+	meterProvider := metric.NewMeterProvider(append(tele.MeterProviderOpts, metric.WithReader(exporter))...)
+	traceProvider, err := traceProvider()
+	if err != nil {
+		return fmt.Errorf("new trace provider: %w", err)
+	}
+
+	go serveMetrics()
+
+	dhtConfig := dht.DefaultConfig()
+	dhtConfig.Mode = dht.ModeOptServer
+	dhtConfig.Logger = slog.New(slog.NewTextHandler(os.Stdout, &slog.HandlerOptions{Level: slog.LevelDebug}))
+	dhtConfig.ProtocolID = protocol.ID(cfg.ProtocolID)
+	dhtConfig.MeterProvider = meterProvider
+	dhtConfig.TracerProvider = traceProvider
+
+	if dhtConfig.ProtocolID == dht.ProtocolIPFS {
+		dhtConfig.Datastore = datastore.NewNullDatastore()
```

Comment:
Don't we always want a Null datastore? What's the behaviour here with a different protocol?